js/mail.js

```diff
@@ -22,6 +22,52 @@ var Mail = {
 			$('#searchresults').hide();
 		}
 	},
+	BackGround: {
+		checkForNotifications: function() {
+			_.each(Mail.State.accounts, function (a) {
+				Mail.Communication.get(OC.generateUrl('apps/mail/accounts/{accountId}/folders', {accountId: a.accountId}), {
+					success: function (jsondata) {
+						var localAccount = Mail.State.folderView.collection.get(jsondata.id);
+						var folders = localAccount.get('folders');
+						_.each(jsondata.folders, function(f) {
+							var localFolder = folders.get(f.id);
+							if (f.uidvalidity !== localFolder.get('uidvalidity') ||
+								f.uidnext !== localFolder.get('uidnext')) {
+
+								// send notification
+								Notification.requestPermission(function() {
+									new Notification(
+										"New Mail arrived!",
```

Comment:
With the current implementation we only know that there was a change on a folder.
And as a result we can reload the current selected folder and update the unread mail counter on all folders.

To get an idea on what is new we'd need to use a different approach .... but doable.